### DISCUSSION TITLE
RBAC based on RC credentials

### DISCUSSION THREAD

**Discussion by rahul101001000** [Sunbird-RC/community#354] [Ideas & Enhancements]
While RC makes it easy to create credentials, we need to also evangelize through product ideas and architectures what it means to use these credentials. One of the important use cases is using credentials to facilitate role based access control (RBAC). 

Here is an attempt to describe what this flow could look like - 

#### User flow
1. Raghavendra is a teacher in a government school and has been issued a 'teacher identity card' credential by the government. The credential contains Raghavendra's Aadhar card number as identifier and his school UDISE. 
2. DIKSHA introduces question contribution and mentoring modules that should only be accessed by teachers. 
3. Typically Raghavendra can login to DIKSHA using phone/OTP and can only access whatever is accessible by everyone. Raghavendra can now import their 'teacher identity card' credential by logging into DigiLocker and consenting to share it with DIKSHA to prove that they are a government school teacher. 
4. Once DIKSHA verifies Raghavendra's 'teacher identity card' credential, Raghavendra gets added to teacher role in DIKSHA. No admin needed to do this. 
5. Voila - role management is fully self serve.

Questions - 
1. Does this make sense?
2. What should be best practices when implementing this? 
3. Should there be a library that makes this easy to implement?

Thanks!

Regards
Rahul

**Comment by pramodkvarma** [comment]
Excellent @rahul101001000, a great example of how registries can allow "one-click" onboarding to various transaction/learning systems.
